src/client/components/app/Transactions/Transaction.tsx

```diff
@@ -37,7 +37,8 @@ interface Asset {
 interface TransactionProps {
   transactionLabel?: string;
   transactionCompleted: boolean;
-  transactionCompletedLabel: string;
+  setTxCompleted?: Dispatch<SetStateAction<boolean>>;
```

Comment:
This is my only concern...

Reply:
1) Type. it is an oddly specific one I know.
2) @xgambitox do you know if we need to clear some state too between txs? Because I saw some TODOS you left about clearing the state in some cases